### ISSUE TITLE
STEPS_PER_EPOCH = No. of images?

### ISSUE THREAD

**Issue by kochmc** [matterport/Mask_RCNN#1645]
Hi there,

I am not sure how best set up STEPS_PER_EPOCH for training my own dataset of 24k images. 

I have set
GPU_COUNT = 1
IMAGES_PER_GPU = 1

So, if I set STEPS_PER_EPOCH=100, for example, does that mean only the first 100 images of my dataset is used for training? So the whole potential of my 24k images would not really be used?

Thanks for help :-)

**Comment by jvendrig** [comment]
I believe your interpretation is correct, but in the second epoch it will use images 101-200 etc., as the data generator will reset only after your whole dataset is exhausted. So after 240 'epochs' you did use the potential of your whole dataset, but they're not truly epochs.
You can set STEPS_PER_EPOCH=24000 instead if that's the size of your training set (excluding validation), and then you get a true epoch. 
The setting depends on how patient you are, as there may be a long time between validation loss computations. I had a similar data set to yours, and set to 2400 as a compromise.
